chia/simulator/block_tools.py

```diff
@@ -795,32 +786,45 @@ def get_consecutive_blocks(
                             else:
                                 pool_target = PoolTarget(self.pool_ph, uint32(0))
 
-                        block_generator: Optional[BlockGenerator]
+                        if dummy_block_references and len(tx_block_heights) > 4:
```

Comment:
Is this code the same just moved around?

Reply:
yes. we have a long standing issue with a lot of code being duplicated in this function. There's a ticket to address it (see comment about test coverage)